lib/qmp.ml

```diff
@@ -149,7 +149,12 @@ let message_of_string str =
 
   let event json =
     let event_data event data =
-      let rtc_offset data = data |> U.member "offset" |> U.to_int in
+      let to_int64 = function
+        | `Int i -> Int64.of_int i
+        | `Intlit i -> Int64.of_string i
```

Comment:
does the qemu code ever send an int, or does it always sent an intlit? (cc: @rosslagerwall)

Reply:
Hi  @mg12, 
QEMU always sends int64.
`qapi_event_send_rtc_change((int64_t)req->data, &error_abort);`
https://code.citrite.net/projects/XS/repos/qemu.pg/commits/4e2e73951f662f24e44f1c812c5c9b187e983190#master/0001-xen-Emit-RTC_CHANGE-upon-TIMEOFFSET-ioreq.patch

Reply:
I think it's good to be overzealous here and allow both int and longint in the unmarshalling.

Reply:
@mg12 What's the difference between int and intlit?

Reply:
It's an internal detail of `Yojson`: http://docs.mirage.io/yojson/Yojson/index.html#type-json.Intlit

For the practical purposes, it uses `Int` for ocaml-compatible integers, and `IntLit` for bigger ones